src/app/pages/bajas/consulta-bajas/consulta-bajas.component.ts

```diff
@@ -7,9 +16,329 @@ import { Component, OnInit } from '@angular/core';
 })
 export class ConsultaBajasComponent implements OnInit {
 
-  constructor() { }
+  settings: any;
+  modo: string; // consulta | revision | aprobacion
+  modoCrud: string; // registrar | ver | editar | revisar | aprobar
+  source: LocalDataSource;
+  estadosMovimiento: Array<EstadoMovimiento>;
+  mostrar: boolean;
+  filaSeleccionada: any;
+  bajaId: number;
+  seleccionados: Array<any>;
+
+  constructor(
+    private confService: ConfiguracionService,
+    private entradasHelper: EntradaHelper,
+    private translate: TranslateService,
+    private pUpManager: PopUpManager,
+    private bajasHelper: BajasHelper,
+    private route: ActivatedRoute,
+  ) { }
 
   ngOnInit() {
+    this.route.data.subscribe(data => {
+      if (data && data.modo !== null && data.modo !== undefined) {
+        this.modo = data.modo;
+      }
+    });
+    this.source = new LocalDataSource();
+    this.loadEstados();
+  }
+
+  public onRegister() {
+    this.modoCrud = 'registrar';
+    this.bajaId = 0;
+  }
+
+  public onEdit(event) {
+    this.filaSeleccionada = event.data;
+    if (this.modo === 'consulta') {
+      if (event.data.EstadoMovimientoId === 'Baja Rechazada') {
+        this.modoCrud = 'editar';
+        this.bajaId = event.data.Id;
+      } else {
+        this.pUpManager.showErrorAlert(this.translate.instant('GLOBAL.bajas.consulta.errorEditar'));
+      }
+    } else if (this.modo === 'revision') {
+      this.modoCrud = 'revisar';
+      this.bajaId = event.data.Id;
+    } else if (this.modo === 'aprobacion') {
+      this.modoCrud = 'aprobar';
+      this.bajaId = event.data.Id;
+    }
```

Comment:
Pareciera que `this.modo` podría ser un candidato a `switch/case` ? Revisar si vale la pena cambiarlo...